src/main/java/de/tum/in/www1/artemis/service/connectors/VersionControlService.java

```diff
@@ -186,4 +188,30 @@ default String getDefaultBranchOfRepository(String projectKey, String repository
      * @return The health of the VCS service containing if it is up and running and any additional data, or the throwing exception otherwise
      */
     ConnectorHealth health();
+
+    /**
+     * Get the default branch used in the participation or retrieves it from the VCS if not present in the database
+     * @param participation The participation to get the default branch from
+     * @return The default branch used by this participation
+     */
+    String getOrRetrieveDefaultBranchOfParticipation(ProgrammingExerciseParticipation participation);
```

Comment:
Can we rename this to `getOrRetrieveBranchOfParticipation`
(same for the methods below)